### DISCUSSION TITLE
options3 ref vs &

### DISCUSSION THREAD

**Discussion by eyyyyyyy3** [rust-lang/rustlings#2029] [Q&A] [answered]
Why is this way of solving the challenge:

```rust
fn main() {
    let y: Option<Point> = Some(Point { x: 100, y: 200 });

    match y {
        Some(ref p) => println!("Co-ordinates are {},{} ", p.x, p.y),
        _ => panic!("no match!"),
    }
    y; // Fix without deleting this line.
}
```

more preferred than this one:
```rust
fn main() {
    let y: Option<Point> = Some(Point { x: 100, y: 200 });

    match &y {
        Some(p) => println!("Co-ordinates are {},{} ", p.x, p.y),
        _ => panic!("no match!"),
    }
    y; // Fix without deleting this line.
}
```

**Comment by sinux-l5d** [comment]
I'm also interesed for feedback. I'm learning Rust and my first though was to `match &y`, before reading about the `ref` keyword.

**Comment by mo8it** [comment, marked as answer]
What do you mean with "more preferred"? Is it because the compiler suggests it?

Personally, I prefer the second solution with `&` over `ref`. `ref` is rarely used in the Rust code that I read until now.

**Reply by eyyyyyyy3** [reply]
Yes. I mean I kinda understand that with ref you can do more advanced pattern matching without deconstructing the tuple on the right hand side. E.g. (ref a, b)
